src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -25,31 +25,53 @@
 package org.jenkinsci.plugins.durabletask;
 
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import hudson.EnvVars;
-import hudson.Extension;
-import hudson.FilePath;
-import hudson.Proc;
-import hudson.Launcher;
+import hudson.*;
 import hudson.util.ListBoxModel;
 import hudson.util.ListBoxModel.Option;
+
+import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import hudson.model.TaskListener;
 import java.io.IOException;
+
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.kohsuke.stapler.DataBoundConstructor;
 import java.io.OutputStream;
 import java.nio.charset.Charset;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import org.kohsuke.stapler.DataBoundSetter;
 
+import javax.annotation.Nonnull;
+
 /**
  * Runs a Powershell script
  */
 public final class PowershellScript extends FileMonitoringTask {
+    @SuppressFBWarnings("MS_SHOULD_BE_FINAL") // Used to control usage of binary or shell wrapper
+    @Restricted(NoExternalUse.class)
+    public static boolean FORCE_BINARY_WRAPPER = Boolean.getBoolean(PowershellScript.class.getName() + ".FORCE_BINARY_WRAPPER");
```

Comment:
side note, I would have expected this to be true so we can remove the old code in a future version (and have this incase compatability is required) - esp if this is required to actually work correctly in environments.

Are we planning on changing this to default to true in the future?

Reply:
You know, for windows I'm perfectly fine with having this become the default right now as this directly addresses a Jenkins bug.

Reply:
I think I'm okay with making this the default for bash as well too. Previously, the binary was not turned on by default just to make sure the old behavior was still working correctly. I'm fairly comfortable now with making the binary default everywhere and the old behavior as a fall-back.

Reply:
FWIW, it might make sense to leave it off by default and do one release just to make sure nothing got messed up in the general refactoring in this PR and other related PRs that have been merged, advertise on the Jenkins mailing lists that the binary exists and will become the default in the future and activate the feature on some actual Jenkins servers that you use to get some real-world testing (maybe you've already done this though), and then do another release that actually turns it back on by default.

(Just thinking back to everything that happened when we tried making the binary the default behavior on linux)

Reply:
Yeah, that's true. I was thinking the pain had already been done on the linux side. But it would be good to make an announcement and then use a SNAPSHOT like you mentioned.

Reply:
using script wrapper by default. see e8a5b455f4ee91b4fd7692fbb16e427ec91ae3e1
will work on a plan to dogfood binary some more and then switch to default in a separate PR